### PR TITLE
[bazel,dvsim] Prevent dvsim from (trying to) copy irrelevant files

### PR DESCRIPTION
**Pull request by pamaury**
See #19553 for details. Due to a bazel change between verions 5 and 6, bazel cquery will print fully qualified paths like @//sw/device/...
for the repository instead of just //sw/device/... This breaks the filtering done when copying file and will make dvsim copy way too many files, or even try to copy files that are not there.
This commit fixes the issue by using a custom priting function for the query that removes the @ for path starting with @//.

### CODE REVIEW

**Comment by pamaury** [comment]
Fixes #19553 and also issues reported on slack/general.

File: hw/dv/tools/dvsim/sim.mk
Comment (cfrantz) [inline hw/dv/tools/dvsim/sim.mk:125]: It would be nice if we could leave a comment nearby explaining this expression.

Unfortunately, this whole block is one giant line continuation.

**Review by cfrantz — APPROVED** [review]

**Review by eshapira — APPROVED** [review]
I'm not familiar with the issue, so let's wait for @timothytrippel approval before merging this PR.

File: hw/dv/tools/dvsim/sim.mk
Comment (pamaury) [reply hw/dv/tools/dvsim/sim.mk:127]: I  just pushed a new version with a comment. I use a fun/terible bash hack to put comments inside a command by using backtrick + comments:
```
echo Hello `# comment` world
```

**Comment by pamaury** [comment]
This has run into the private CI time limit, not sure if this is related to this change?

**Comment by timothytrippel** [comment]
I think this should be OK, but it looks like the OTBN tests don't pass, which is odd, because this change should only impact top-level tests (which do all pass). But I will re-trigger that CI step to make sure this passes on everything before we merge. If it does pass CI then we should be OK to merge.

Screenshot 2023-09-05 at 3 16 45 PM

Thanks for catching this @pamaury

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**
